File: 3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.java
class Solution{
    public int solve(int a, int b){
        if(b == 0)
            return a;
        return solve(b, a%b);
    }

    public int gcdOfOddEvenSums(int n){
        int odd=(n*n);
        int even=n*(n+1);

        return solve(odd, even);

    }
}

File: Easy/3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.py
class Solution:
    def solve(self, a, b):
        if b == 0:
            return a
        return self.solve(b, a%b)
    
    def gcdOfOddEvenSums(self, n: int) -> int:
        odd=n*n
        even=n*(n+1)

        return self.solve(odd, even)